lib/rules/object-shorthand.js

```diff
@@ -497,6 +497,13 @@ module.exports = {
                         node,
                         messageId: "expectedPropertyShorthand",
                         fix(fixer) {
+                            const lastKeyToken = sourceCode.getLastToken(node.key);
+
+                            // x: /* */ x
+                            if (sourceCode.commentsExistBetween(lastKeyToken, node.value)) {
+                                return null;
+                            }
```

Comment:
```suggestion

                            // x: /* */ x
                            if (sourceCode.commentsExistBetween(node.key, node.value)) {
                                return null;
                            }
```

We can use `node.key` directly.

Reply:
[d5c6fe1](https://github.com/eslint/eslint/pull/18442/commits/d5c6fe1f8540f299d010e1256eff97151b995e42)